umm-spec-lib/test/cmr/umm_spec/test/umm_json.clj

```diff
@@ -125,6 +125,14 @@
           parsed (remove-get-service-and-get-data-nils parsed)]
       (is (= umm-s-record parsed)))))
 
+(deftest all-umm-t-records
+  (checking "all umm-t records" 100
+    [umm-t-record (gen/no-shrink umm-gen/umm-t-generator)]
+    (let [json (umm-json/umm->json umm-t-record)
+          _ (is (empty? (json-schema/validate-umm-json json :tool)))
```

Comment:
Is the test validating the generator output or the parser?

Reply:
generator output, where ToolKeywords are nil sometimes, even though it's required field in the schema

Reply:
Sorry, both. and indirectly testing the schema. In this case, ToolKeywords minItems was 0 which is wrong so we fixed it.